### ISSUE TITLE
Bind variable that includes in its name the work 'hashedOldPassword' does not parse correctly

### ISSUE THREAD

**Issue by tkallioras** [arangodb/arangodb#5738]
## My Environment

* __ArangoDB Version__:        3.3.11 enterprise
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Infrastructure__:               own
* __Operating System__:        MacOS 10.14 beta 
* __Used Package__:              Docker - official Docker library

Whenever a bind variable is named **hashedOldPassword** the query does not return correct results.

**For example** (with the word **hashedOldPassword**):

```
Query string:
 LET user = FIRST(
 FOR u IN @@usersCollection
 	FILTER u._id == @userID && u.primaryEmail == @email && u.password == @hashedOldPassword
 	RETURN u)
 	
 RETURN user != null

Execution plan:
 Id   NodeType          Site       Est.   Comment
  1   SingletonNode     COOR          1   * ROOT
  7   SubqueryNode      COOR          1     - LET #1 = ...   /* const subquery */
  2   SingletonNode     DBS           1       * ROOT
 11   IndexNode         DBS           1         - FOR u IN users   /* primary index scan */
 12   CalculationNode   DBS           1           - LET #4 = ((u.`primaryEmail` == "test@test.com") && (u.`password` == "randomHashedPassword"))   /* simple expression */   /* collections used: u : users */
  5   FilterNode        DBS           1           - FILTER #4
 15   RemoteNode        COOR          1           - REMOTE
 16   GatherNode        COOR          1           - GATHER 
  6   ReturnNode        COOR          1           - RETURN u
  9   CalculationNode   COOR          1     - LET #6 = (FIRST(#1) != null)   /* simple expression */
 10   ReturnNode        COOR          1     - RETURN #6

Indexes used:
 By   Type      Collection   Unique   Sparse   Selectivity   Fields       Ranges
 11   primary   users        true     false            n/a   [ `_key` ]   (u.`_id` == "users/0b3b564e-0525-595a-804e-d8ecc379ae1a")

Optimization rules applied:
 Id   RuleName
  1   remove-unnecessary-calculations
  2   use-indexes
  3   remove-filter-covered-by-index
  4   scatter-in-cluster
  5   distribute-filtercalc-to-cluster
  6   remove-unnecessary-remote-scatter
```

**Result:** 

```
[
  false
]
```

Whereas without the word **hashedOldPassword**:

```
Query string:
 LET user = FIRST(
 FOR u IN @@usersCollection
 	FILTER u._id == @userID && u.primaryEmail == @email && u.password == @oldPassword
 	RETURN u)
 	
 RETURN user != null

Execution plan:
 Id   NodeType          Site       Est.   Comment
  1   SingletonNode     COOR          1   * ROOT
  7   SubqueryNode      COOR          1     - LET #1 = ...   /* const subquery */
  2   SingletonNode     DBS           1       * ROOT
 11   IndexNode         DBS           1         - FOR u IN users   /* primary index scan */
 12   CalculationNode   DBS           1           - LET #4 = ((u.`primaryEmail` == "test@test.com") && (u.`password` == "randomHashedPassword"))   /* simple expression */   /* collections used: u : users */
  5   FilterNode        DBS           1           - FILTER #4
 15   RemoteNode        COOR          1           - REMOTE
 16   GatherNode        COOR          1           - GATHER 
  6   ReturnNode        COOR          1           - RETURN u
  9   CalculationNode   COOR          1     - LET #6 = (FIRST(#1) != null)   /* simple expression */
 10   ReturnNode        COOR          1     - RETURN #6

Indexes used:
 By   Type      Collection   Unique   Sparse   Selectivity   Fields       Ranges
 11   primary   users        true     false            n/a   [ `_key` ]   (u.`_id` == "users/0b3b564e-0525-595a-804e-d8ecc379ae1a")

Optimization rules applied:
 Id   RuleName
  1   remove-unnecessary-calculations
  2   use-indexes
  3   remove-filter-covered-by-index
  4   scatter-in-cluster
  5   distribute-filtercalc-to-cluster
  6   remove-unnecessary-remote-scatter
```

**Result:** 

```
[
  true
]
```

**Comment by jsteemann** [comment]
I don't think this is related to the naming of the bind variable.
The execution plans for both queries look absolutely identical.
And the string "hashedOldPassword" is not used by ArangoDB itself in any context.

It rather seems this issue could be a side-effect of something else going wrong, e.g. https://github.com/arangodb/arangodb/issues/5727

**Comment by tkallioras** [comment]
I am running a different AQL query, using again bind variables. The query and the explanation are the following:

```
Query string:
 WITH @@usersCollection
 FOR v, e, p
 IN 1..1
 INBOUND @userID
 GRAPH @friendRequestsGraph
 OPTIONS { uniqueVertices: 'path' }
 			
 FILTER p.vertices[1]._id == @friendID
 		
 LET friendEdges = [@edge1, @edge2]
 FOR friendEdge IN friendEdges
 INSERT friendEdge IN @@friendsEdge
 REMOVE p.edges[0] IN @@friendRequestsEdge
 		
 RETURN 1

Execution plan:
 Id   NodeType            Site       Est.   Comment
  1   SingletonNode       COOR          1   * ROOT
  5   CalculationNode     COOR          1     - LET friendEdges = [ "", "" ]   /* json expression */   /* const assignment */
 11   CalculationNode     COOR          1     - LET #15 = 1   /* json expression */   /* const assignment */
  2   TraversalNode       COOR          1     - FOR v  /* vertex */, p  /* paths */ IN 1..1  /* min..maxPathDepth */ INBOUND 'users/42a25854-4f6c-57ee-ada2-4e65073dd42f' /* startnode */  GRAPH 'friendRequestsGraph'
  9   CalculationNode     COOR          1     - LET #13 = p.`edges`[0]   /* simple expression */
  7   EnumerateListNode   COOR          2     - FOR friendEdge IN friendEdges   /* list iteration */
 17   ScatterNode         COOR          2       - SCATTER
 18   RemoteNode          DBS           2       - REMOTE
  8   InsertNode          DBS           0       - INSERT friendEdge IN friendsEdge
 19   RemoteNode          COOR          0       - REMOTE
 20   GatherNode          COOR          0       - GATHER 
 13   DistributeNode      COOR          0       - DISTRIBUTE
 14   RemoteNode          DBS           0       - REMOTE
 10   RemoveNode          DBS           0       - REMOVE #13 IN friendRequestsEdge
 15   RemoteNode          COOR          0       - REMOTE
 16   GatherNode          COOR          0       - GATHER 
 12   ReturnNode          COOR          0       - RETURN #15

Indexes used:
 By   Type   Collection           Unique   Sparse   Selectivity   Fields      Ranges
  2   edge   friendRequestsEdge   false    false            n/a   [ `_to` ]   base INBOUND

Traversals on graphs:
 Id   Depth   Vertex collections   Edge collections     Options                                   Filter conditions
  2   1..1    users                friendRequestsEdge   uniqueVertices: path, uniqueEdges: path   (p.`vertices`[1].`_id` == "users/ea514435-619e-564c-a420-ff19b001b7e9")

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   move-filters-up
  3   remove-redundant-calculations
  4   remove-unnecessary-calculations
  5   move-calculations-up-2
  6   move-filters-up-2
  7   remove-data-modification-out-variables
  8   optimize-traversals
  9   remove-filter-covered-by-traversal
 10   remove-unnecessary-calculations-2
 11   distribute-in-cluster
 12   scatter-in-cluster

Write query options:
 Option                   Value
 ignoreErrors             false
 waitForSync              false
 nullMeansRemove          false
 mergeObjects             true
 ignoreDocumentNotFound   false
 readCompleteInput        true
 useIsRestore             false
 consultAqlWriteFilter    false
```

However I get this error:

`AQL: Error message received from shard '' on cluster node 'PRMR-565f1484-128e-4e8b-b67d-633b7a872ce3': document not found (while executing)`

After analyzing the query I isolated the issue when the server tries to remove the last edge from `friendRequestsEdge`

_However, when I remove the bind variable and I execute the query using the collection directly in the AQL, the query runs as expected_

Could those issues be related somehow?

**Comment by jsteemann** [comment]
I don't think the bind variable plays any role here.
I think there is something wrong with the distribution of the documents to the shards in some case, as indicated by #5727. This leads to arbitrary follow-up failures.
So we need to fix #5727. Work on this is under way already.

**Comment by vinaypyati** [comment]
Hi @tkallioras ,

Did you have a chance to test this on the latest version(3.3.13) ? Does this issue still persists? 

Thanks!

**Comment by tkallioras** [comment]
It seems to be solved in version 3.3.13
